pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java

```diff
@@ -30,9 +30,38 @@
   /**
    * Return whether the client has data access to the given table.
    *
-   * @param httpHeaders Http headers
+   * Note: This method is only used fore read access. It's being deprecated and its usage will be replaced by
+   * `hasAccess` method with AccessType.READ.
+   *
+   * @param httpHeaders HTTP headers containing requester identity
    * @param tableName Name of the table to be accessed
    * @return Whether the client has data access to the table
    */
+  @Deprecated
   boolean hasDataAccess(HttpHeaders httpHeaders, String tableName);
+
+  /**
+   * Return whether the client has permission to the given table
+   *
+   * @param tableName name of the table to be accessed
+   * @param accessType type of the access
+   * @param httpHeaders HTTP headers containing requester identity
+   * @param endpointUrl the request url for which this access control is called
+   * @return whether the client has permission
+   */
+  default boolean hasAccess(String tableName, AccessType accessType, HttpHeaders httpHeaders, String endpointUrl) {
```

Comment:
We can do one of:
(1) Have a default method here that returns "true"
(2) Have a default implementation of the interface that always return true. Actual implementations need to override the method to return the value they want.

Earlier, we followed the pattern (1). Since we are deprecating the first API, we can eventually remove the default implementations and follow pattern (2) moving forward. I prefer that.

Either way, for installations that are not interested in authenticating, we must come up by default instead of requiring them to implement an empty auth class

Reply:
You both are right, the default values in the interface is for smooth transition. When the first api gets removed, the default values in the interface will be removed as well and the default implementation of the interface will return true for both methods.